code/modules/mob/living/basic/lavaland/goliath/goliath_actions.dm

```diff
@@ -0,0 +1,83 @@
+/// Place some grappling tentacles underfoot
+/datum/action/cooldown/goliath_tentacles
+	name = "Unleash Tentacles"
+	desc = "Unleash burrowed tentacles at a targetted location, grappling targets after a delay."
+	button_icon = 'icons/mob/simple/lavaland/lavaland_monsters.dmi'
+	button_icon_state = "goliath_tentacle_wiggle"
+	background_icon_state = "bg_demon"
+	overlay_icon_state = "bg_demon_border"
+	click_to_activate = TRUE
+	cooldown_time = 12 SECONDS
+	melee_cooldown_time = 0
+	check_flags = AB_CHECK_CONSCIOUS | AB_CHECK_INCAPACITATED
+	/// Furthest range we can activate ability at
+	var/max_range = 7
+
+/datum/action/cooldown/goliath_tentacles/Activate(atom/target)
+	target = get_turf(target)
+	if (get_dist(owner, target) > max_range)
+		return FALSE
+
+	. = ..()
```

Comment:
is there really no proc like `PreActivate()` where this could be passed in? Pretty silly to have the parent call in the middle proc like this in modern code imo, but that's an issue to implement another time. just making note of it now